python/tank/platform/engine.py

```diff
@@ -2677,7 +2677,37 @@ def _start_engine(engine_name, tk, old_context, new_context):
             LogManager().initialize_base_file_handler(engine_name)
 
         # get environment and engine location
-        (env, engine_descriptor) = get_env_and_descriptor_for_engine(engine_name, tk, new_context)
+        try:
+            (env, engine_descriptor) = get_env_and_descriptor_for_engine(engine_name, tk, new_context)
+        except TankEngineInitError:
+            # If we failed using the typical pick_environment approach, then we
+            # need to check and see if this is the Shotgun engine. If it is, then
+            # we can also try the legacy engine start method, which will make use
+            # of shotgun_xxx.yml environment files if they exist in the config.
+            if engine_name == constants.SHOTGUN_ENGINE_NAME:
+                # If the new context has an associated entity, we get the type from
+                # that. If it doesn't, then we can check to see if it's a project
+                # context. If neither is the case, meaning we have an empty context,
+                # then we re-raise.
+                if new_context.entity is not None:
+                    entity_type = new_context.entity.get("type")
+
+                    # This should be exceedingly unlikely, and if it occurs is
```

Comment:
how is this possible at all? would be great with some more info! I don't think we have ever allowed for `context.entity` to exist without a type key! So maybe just `entity_type = new_context.entity["type"]`